Terminal.Gui/Views/ComboBox.cs

```diff
@@ -33,7 +28,11 @@ public class ComboBox : View {
 			get => source;
 			set {
 				source = value;
-				SetNeedsDisplay ();
+
+				if(isAdded) { 
```

Comment:
Can you explain why `isAdded` is needed? I don't understand it. There's probably a good reason, but it's not clear to me.

Note you never set it to false.

If it's really needed, you can use `if (SuperView != null && SuperView.SubViews.IndexOf(this) != -1)` to accomplish the same thing; this will also protect against the case where the view has been added twice.

Reply:
Thanks @tig Its for the case where the source is set **before** the control is added. There is no need to force it to refresh as this will be triggered by  the call to Add()

```csharp
comboBox = new ListView();
comboBox.SetSource(list);
mainWindow.Add(comboBox);
```

Reply:
Implemented